service/endpoint.cc

```diff
@@ -64,6 +64,16 @@ EndpointBase::
 
 void
 EndpointBase::
+setPollingMode(enum PollingMode mode)
```

Comment:
Because there are more than 2 modes. An enum for polling modes seemed the easier way to segregate modes without risking to interfere with each of them. Specifying "modes" or "profiles" separates them entirely from each other.